### PR TITLE
Filtering PointCloud into Obstacles PointCloud

### PR DESCRIPTION
**Pull request by staylo32**
working PointCloud filter that extracts out z values between certain values and publishes the output.

### CODE REVIEW

**Comment by l0g1x** [comment]
Can one of the admins verify this patch?

File: pcl_ground_filter/package.xml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0"?>
+<package>
+  <name>pcl_ground_filter</name>
+  <version>0.0.0</version>
+  <description>The pcl_ground_filter package</description>
+
+  <!-- One maintainer tag required, multiple allowed, one person per tag --> 
+  <!-- Example:  -->
+  <!-- <maintainer email="jane.doe@example.com">Jane Doe</maintainer> -->
+  <maintainer email="scipio@todo.todo">scipio</maintainer>
```
Comment (l0g1x) [inline pcl_ground_filter/package.xml]: @staylo32 can you fill this info out?

File: pcl_ground_filter/src/pcl_ground_filter.cpp [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+#include <sensor_msgs/PointCloud2.h>
+#include <sensor_msgs/PointCloud.h>
+#include <sensor_msgs/point_cloud_conversion.h>
+#include <tf/transform_listener.h>
+
+#include <iostream>
+#include <pcl/ModelCoefficients.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/point_types.h>
+#include <pcl/sample_consensus/method_types.h>
```
Comment (l0g1x) [inline pcl_ground_filter/src/pcl_ground_filter.cpp]: Are we using still using these libraries with no ransac?

File: pcl_ground_filter/src/pcl_ground_filter.cpp [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+  filterByHeight(pc, min_z, max_z);
+  PCL2 nongroundPCL2;
+
+  pcl::toPCLPointCloud2(pc,nongroundPCL2); // convert from pcl::PointCloud to pcl::PCLPointCloud2
+
+  // Publish the nonground pc to obstacles
+  publishPCLPointCloud(nongroundPCL2, obstacles_pub_);
+}
+/* Takes in sensor_msgs::PointCloud2, converts it to sensor_msgs::PointCloud2, performs transform,
```
Comment (bsubei) [inline pcl_ground_filter/src/pcl_ground_filter.cpp:95]: are these comments correct? I can't understand them

File: nmea_navsat_driver [hunk 4]
```diff
@@ -1 +0,0 @@
-Subproject commit 24a2275006bab31ecbcebf2d05be5f0ca8fdd643
```
Comment (bsubei) [inline nmea_navsat_driver:1]: gotta remove this change. Can you please amend this commit and remove this change? We want to keep this nmea_navsat_driver submodule file.

**Comment by bsubei** [comment]
good work. Sounds like it needs dynamic_reconfigure to change the z-height cutoff value... ;)

**Comment by l0g1x** [comment]
Yes i was going to say this would be the next thing steven can implement for min_z and max_z

**Comment by l0g1x** [comment]
test this please

**Pull request merged**